toolbox/metrics/buildFreshness/buildFreshness.go

```diff
@@ -35,27 +35,27 @@ type DepFreshness struct {
 	Age time.Duration
 }
 
-func getBranchHeadTime(githubClnt *u.GithubClient, repo, branch string) (*time.Time, error) {
+func getBranchHeadTime(githubClnt *u.GithubClient, repo, branch string) (time.Time, error) {
 	sha, err := githubClnt.GetHeadCommitSHA(repo, branch)
 	if err != nil {
-		return nil, err
+		return time.Time{}, err
```

Comment:
Is there a reason why we want to switch to returning stucts from returning pointers?
Returning structs basically copies the entire struct from the stack frame of the current function to its caller. Performance aside, finding every occurrence of this functions (various of others also being refactored) can be challenging and prone to bugs. Perhaps running `bazel build //...` helps catching some of them.
I am not sure about this one. Perhaps @sebastienvas may have a better take at it.

Reply:
The idea comes from using method `Getxxx()` instead of directly access a member. I think that's 
the safer way get value. But I am kind of curious about why they still make those member public though.

Reply:
who are they?

Reply:
I don t think performance is an issue here. You must return a value when you don't want it to be modified. When I was working with this github library I had to redo my function signature multiple time. I think using value is fine

Reply:
They: github golang api